### PR TITLE
[prometheus] Add support for external_labels

### PR DESCRIPTION
**Pull request by adamchabin**
#### What this PR does / why we need it
Support for external_labels option under global section.

#### Which issue this PR fixes
When you use multiple Prometheus as agents you would like to add tags like tenant:

external_labels:
   tenant: 'tenant1'

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: prometheus
 appVersion: v2.47.0
-version: 25.1.0
+version: 25.1.1
```
Comment (zeritti) [inline charts/prometheus/Chart.yaml:4]: Thank you, @adamchabin, for your PR. Please, bump the minor number with a new feature.

```suggestion
version: 25.2.0
```

File: charts/prometheus/templates/cm.yaml [hunk 2]
```diff
@@ -20,6 +20,10 @@ data:
 {{- if eq $key "prometheus.yml" }}
     global:
 {{ $root.Values.server.global | toYaml | trimSuffix "\n" | indent 6 }}
+{{- if $root.Values.server.external_labels }}
+      external_labels:
+{{ $root.Values.server.external_labels | toYaml | indent 8 }}
+{{- end }}
```
Comment (zeritti) [inline charts/prometheus/templates/cm.yaml:26]: I'd suggest to name the field with camelCase as usual, i.e. `externalLabels`.

```suggestion
{{- with $root.Values.server.externalLabels }}
      external_labels: {{ toYaml . | nindent 8 }}
{{- end }}
```

File: charts/prometheus/values.schema.json [hunk 3]
```diff
@@ -271,6 +271,9 @@
                 "global": {
                     "type": "object",
                     "properties": {
+                        "external_labels": {
+                           "type": "object"
+                        },
```
Comment (zeritti) [inline charts/prometheus/values.schema.json:276]: As shown above, you are setting the field nested in `server`, not in `global`. The schema has to include `server.externalLabels` instead. Please, remember to insert the new field in `values.yaml` as well with a short comment/reference as seen e.g. at `server.remoteRead`.

**Review by zeritti — CHANGES_REQUESTED** [review]

File: charts/prometheus/Chart.yaml (hunk 1 above)
Comment (adamchabin) [reply charts/prometheus/Chart.yaml:4]: Agreed

File: charts/prometheus/templates/cm.yaml (hunk 2 above)
Comment (adamchabin) [reply charts/prometheus/templates/cm.yaml:26]: Agreed

**Review by zeritti — CHANGES_REQUESTED** [review]

**Pull request closed without merging**